internal/daemonset/daemonset.go

```diff
@@ -0,0 +1,200 @@
+		},
+	}
+}
+
+// Delete dameno set
+func DeleteDaemonSet(daemonSetName, namespace string) error {
+	tnf.ClaimFilePrintf("Deleting daemon set %s\n", daemonSetName)
+	deletePolicy := metav1.DeletePropagationForeground
+	client := clientsholder.GetClientsHolder().K8sClient.AppsV1()
+	if err := client.DaemonSets(namespace).Delete(context.TODO(), daemonSetName, metav1.DeleteOptions{
+		PropagationPolicy: &deletePolicy,
+	}); err != nil {
+		tnf.ClaimFilePrintf("The daemonset (%s) deletion is unsuccessful due to %s", daemonSetName, err)
+	}
+	doneCleanUp := false
+	for start := time.Now(); !doneCleanUp && time.Since(start) < timeout; {
+		client := clientsholder.GetClientsHolder().K8sClient
+		pods, err := client.CoreV1().Pods(namespace).List(context.TODO(), metav1.ListOptions{LabelSelector: "name=" + daemonSetName})
+		if err != nil {
+			return fmt.Errorf("failed to get pods, err: %s", err)
+		}
+		//nolint:ineffassign
+		if len(pods.Items) == 0 {
+			doneCleanUp = true
+			break
+		}
+		time.Sleep(time.Duration(timeout.Minutes()))
+	}
+
+	tnf.ClaimFilePrintf("Successfully cleaned up daemon set %s\n", daemonSetName)
+	return nil
+}
+
+// Check if the daemon set exists
+func doesDaemonSetExist(daemonSetName, namespace string) bool {
+	client := clientsholder.GetClientsHolder().K8sClient.AppsV1()
+	_, err := client.DaemonSets(namespace).Get(context.TODO(), daemonSetName, metav1.GetOptions{})
+	if err != nil {
+		tnf.ClaimFilePrintf("Error occurred checking for Daemonset to exist: " + err.Error())
+	}
+	// If the error is not found, that means the daemon set exists
+	return err == nil
+}
+
+// Create daemon set
+func CreateDaemonSet(daemonSetName, namespace, containerName, imageWithVersion string, timeout time.Duration) (*corev1.PodList, error) {
+	rebootDaemonSet := CreateDaemonSetsTemplate(daemonSetName, namespace, containerName, imageWithVersion)
+	if doesDaemonSetExist(daemonSetName, namespace) {
+		err := DeleteDaemonSet(daemonSetName, namespace)
+		if err != nil {
+			tnf.ClaimFilePrintf("Failed to delete debug daemonset because: %s", err)
+		}
+	}
+
+	tnf.ClaimFilePrintf("Creating daemon set %s\n", daemonSetName)
+	client := clientsholder.GetClientsHolder().K8sClient
+	_, err := client.AppsV1().DaemonSets(namespace).Create(context.TODO(), rebootDaemonSet, metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+	err = provider.WaitDebugPodsReady()
+	if err != nil {
+		return nil, err
+	}
+
+	tnf.ClaimFilePrintf("Daemonset is ready")
+
+	var ptpPods *corev1.PodList
+	ptpPods, err = client.CoreV1().Pods(namespace).List(context.TODO(), metav1.ListOptions{LabelSelector: "name=" + daemonSetName})
+	if err != nil {
+		return ptpPods, err
+	}
+	tnf.ClaimFilePrintf("Successfully created daemon set %s\n", daemonSetName)
+	return ptpPods, nil
+}
+
+// Deploy daemon set on repo partner
+func PartnerRepoDaemonset() map[string]corev1.Pod {
+	dsRunningPods, err := CreateDaemonSet(daemonSetName, namespace, containerName, imageWithVersion, timeout)
```

Comment:
Use provider.DaemonSetName here, otherwise it will not match the name used by the provider.WaitDebugPodsReady function (debug!=partner-repo)